### PR TITLE
Updating `sub` for regression test

### PR DESCRIPTION
**Pull request by ludomal**
`sub` executable was updated to return an error code when the files being compared are different.
`sub` is the tool that compare files for bit-exactness or equivalence. It should be used in the regression tests instead of `cf`.

### CODE REVIEW

File: src/unsup/sub-add.c [hunk 1]
```diff
@@ -270,6 +284,12 @@ int main (int argc, char *argv[]) {
     printf ("(%ld equivalent at a +-%ld level) ", NrEquivs, equiv);
   printf ("found out of %ld.\n", N * N2);
 
+  if (return_error) {
```
Comment (othiman) [inline src/unsup/sub-add.c]: Maybe a dump question, but the sub-add function does not return any errors if the `-e` option is not used, does it? Is this intended?

Comment (ludomal) [reply src/unsup/sub-add.c]: Correct. The original code would display that the files are different but would not return an error.

This patch provides a switch to return an error if the files are different. This makes scripting much easier. This is provided as an option for backward compatibilities.

Comment (othiman) [reply src/unsup/sub-add.c]: Sounds good. :-)

**Comment by dennisguse** [comment]
Since we are now starting to use `sub` for testing, it would be best to remove it from unsupported to support.
What do you think?

**Comment by dennisguse** [comment]
@ludomal I updated the PR according to my suggestions.
What do you think?

**Comment by ludomal** [comment]
Agreed, I think it is ok to loose backward compatibility in this tool. Thanks for updating the code.
Let's merge it. I will update the remaining pull requests that use this tool.

**Comment by dennisguse** [comment]
@ludomal Great. Could you review and approve the code?
I am not really into reviewing my own mistakes ;)
When approved, I will squash commit it.

And what about moving `sub` from `unsup` to `utl`?

**Comment by ludomal** [comment]
I have looked at the code and I am happy with it but I cannot approve it as the initial pull request is mine.

**Pull request merged**